File: 55-JumpGame/55-JumpGame.py
# Last updated: 6/11/2026, 11:37:40 AM
class Solution:
    def canJump(self, nums: List[int]) -> bool:
        win_post = len(nums)-1
        for i in range(len(nums)-2,-1,-1):
            if(i+nums[i] >= win_post):
                win_post = i
        return win_post == 0              

File: 55-JumpGame/55-JumpGame.cpp
// Last updated: 6/11/2026, 11:37:34 AM
class Solution {
public:
    bool canJump(vector<int>& nums) {
        int n = nums.size();
        int mxl = 0;
        for(int i=0; i<n-1; i++){
            mxl = max(mxl, i+nums[i]);
            if(mxl == i) return 0;
        }
        return 1;
    }
};